### DISCUSSION TITLE
Apple 27" LED Cinema/Thunderbolt Displays permanently dimmed at ~30%.

### DISCUSSION THREAD

**Discussion by chazzzzy** [MonitorControl/MonitorControl#739] [Q&A] [answered]
I just installed Monterey on my Mac mini M1 with 2 Apple 27" LED cinema Displays and I noticed that there is no longer a way to adjust the brightness, they were stuck on Maximum Brightness. (This was/is a MONTEREY issue NOT MonitorControl)

But then I made the mistake of installing MonitorControl (4.0.0-rc1) to see if I could adjust the brightness that way and upon running the app it immediately dimmed the display hooked up via USB-C and a mini displayport adapter to about 30% maximum brightness and it has remained that way ever since, even after quitting the app and restarting the computer. I can dim the display all the way to black if I want via MonitorControl, but maximum brightness is now stuck at around 30%.

My display hooked up to the HDMI port remained bright.  

So I decided to alternate how the displays were connected, hooking the HDMI one to the USB-C, and vice-versa.. and now BOTH displays are permanently dimmed at 30% max!

Both displays now have a maximum brightness of 30% even after restarting the computer, disabling MonitorControl, unplugging and plugging in the displays, etc.. 

Does anyone have any ideas on what I can do to reset them to maximum brightness? It seems like the displays are INTERNALLY stuck at 30% brightness and MonitorControl thinks the 30% is the maximum brightness.

Perhaps I need to plug them into another computer and brighten them that way and then plug them back into my computer with MonitorControl not running?

Or is there a way to make MonitorControl allow me to brighten them beyond the 30% limitation that it somehow triggered upon running?

Thanks so much!

Charles

**Comment by chazzzzy** [comment, marked as answer]
I solved the permanently dimmed screens issue. Here is how I did it (taken from here: [https://discussions.apple.com/thread/252063309?answerId=255777656022#255777656022](url) )

This works in Monterey as well.. doesn't require Big Sur.

I downloaded the source code of v3.0.0-rc1 of MonitorControl from here: [https://github.com/MonitorControl/MonitorControl/releases/tag/v3.0.0-rc1](url)

Then opened the folder up in Xcode.

Did a search for: DDC_HARD_MAX_LIMIT and changed it from 100 to 250

Then RUN the code

It asks you to allow it in the Security Preferences.. I just unchecked then checked it again there, since I still had other version of MonitorControl on my computer.

I was then able to adjust the brightness of each display to the maximum brightness! 

I could probably do this in a later version of the source code, but I wanted to do it on a version that someone else had done it on just to be safe.

Now my displays are no longer permanently dimmed! Yay!

**Reply by waydabber** [reply]
Amazing, great find! It is actually rare that a display needs anything higher than 100 as max DDC brightness, but Apple displays are always special (normally they are not controlled via DDC but via Apple's own CoreDisplay services via an USB based protocol).

If you wish to use v4.0.0 instead, you can overwrite the max DDC brightness as there is a setting to that now for situations like this:

<img width="842" alt="Screen Shot 2021-10-26 at 18 42 13" src="https://user-images.githubusercontent.com/37590873/138923583-411f17b9-994c-4de7-b03d-0a8c525205d8.png">

(edit: I think its better to use value 255)

**Reply by chazzzzy** [reply]
Thanks so much @waydabber! That is very helpful! I'll be installing MonitorControl v. 4.0.0 on my computer as we speak!

**Reply by waydabber** [reply]
Hopefully it works... :)) Let me know if it doesn't, I didn't really have a chance to test it on more exotic displays.

**Reply by nikkimoreaux** [reply]
@waydabber No more dimming problem! This 250 DDC max setting solved the Monterey bug with my configuration! (M1 Mac Mini + MacOS 12.0.1 + a good old 27" LED Cinema Display 2010 connected thru a cheap noname USB-C > Mini-displayport dongle)
Thank you! **A LOT!!!**

Hardware Overview:
  Model Name:	Mac mini
  Model Identifier:	Macmini9,1
  Chip:	Apple M1
  Total Number of Cores:	8 (4 performance and 4 efficiency)
  Memory:	16 GB
  System Firmware Version:	7429.41.5
  OS Loader Version:	7429.41.5
  Activation Lock Status:	Enabled

LED Cinema Display:
  Resolution:	2560 x 1440 (QHD/WQHD - Wide Quad High Definition)
  UI Looks like:	2560 x 1440 @ 60.00Hz
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	Yes

System Software Overview:
  System Version:	macOS 12.0.1 (21A559)
  Kernel Version:	Darwin 21.1.0
  Boot Volume:	Macintosh HD
  Boot Mode:	Normal
  Computer Name:	mini
  User Name:	Nikki Moreaux (nikki)
  Secure Virtual Memory:	Enabled
  System Integrity Protection:	Enabled
  Time since boot:	23:25


![Screen Shot 2021-10-27 at 05 40 22](https://user-images.githubusercontent.com/1276977/138996064-a57d0b05-02fa-403b-94f6-cc0b144ec665.png)

**Reply by waydabber** [reply]
Glad it works, thanks for the feedback! :)

**Reply by waydabber** [reply]
(btw you can probably use 255 instead of 250 - let me know if that works as well, that should be a more logical as it corresponds to 0xFF the max value representable in one byte)

**Reply by dansure5** [reply]
Thanks a lot! 255 is the better value. Anything below that isn‘t maximum, anything beyond that marks the display darker and the volume lower.

**Reply by dansure5** [reply]
@waydabber I am using the same setup as you Mini Display to USB-C adapter on M1 Mac Mini. Everything works perfectly, but I can‘t get the microphone to work on FaceTime. Camera works perfectly fine though. Do you have the same issue?

**Reply by pajai** [reply]
Trying to fix this external Cinema Display stuck at 40% brightness.

I tried to install MonitorControl (latest version, 4.1.0), but I am not able to reproduce the fix. I am doing probably something wrong. I am on Monterey 12.4 with a 14'' macbook pro.

Any hint what I am doing wrong?

Here are my settings:

![Screenshot 2022-06-18 at 01 01 24](https://user-images.githubusercontent.com/490037/174410823-b7cc5026-3a61-48a7-91d8-886dca6426de.png)

![Screenshot 2022-06-18 at 01 01 06](https://user-images.githubusercontent.com/490037/174410843-92626599-b909-465b-9026-0ebd63c48fba.png)

**Reply by dansure5** [reply]
> Amazing, great find! It is actually rare that a display needs anything higher than 100 as max DDC brightness, but Apple displays are always special (normally they are not controlled via DDC but via Apple's own CoreDisplay services via an USB based protocol).
> 
> If you wish to use v4.0.0 instead, you can overwrite the max DDC brightness as there is a setting to that now for situations like this:
> 
> <img alt="Screen Shot 2021-10-26 at 18 42 13" width="842" src="https://user-images.githubusercontent.com/37590873/138923583-411f17b9-994c-4de7-b03d-0a8c525205d8.png">
> 
> (edit: I think its better to use value 255)

Try setting the minimum value to 0.

**Reply by pajai** [reply]
On my computer, it has no effect.

Weirdly, I have the impression that the DDC max has no effect either. When I change it to 50 or even 10, the max brightness is still the same. Am I supposed to save or activate a setting?

**Reply by ioannoukyr** [reply]
I have the same issue as @pajai under Monterey 12.4, and in addition to that changing the brightness will sometimes cause the monitor to lose signal and show B/W static noise which i can fix by replugging the monitor.

**Comment by scroggzen** [comment]
I tried using the method described above on my Apple 27" Thunderbolt display to adjust the DDC max to 250... and it still isn't restoring the previous brightness. I'm getting the same results on both a Mojave MBPro direct thunderbolt in and a Big Sur MBPro with Thunderbolt to USB-C

Although I am changing the DDC max to 250, it appears to be stuck at the ~50% brightness mark. 0-50% dimming works, but beyond that it simply stops doing anything from ~50% - 100%

I've tried unchecking "Use hardware DDC control" as well as unchecking "Combine hardware and software dimming" and have the same results whether using hardware controls or the software slider. I've also tried checking off and on "Sync brightness changes from Built-in and Apple displays" and this generally just makes the touch bar work when on and not when off.

Basically all functionality is fine, it just perma-nerfed the brightness max to 50%, and I'm wondering why the same solution that's working for others isn't working for me.

**Reply by waydabber** [reply]
Hi @scroggzen - Did MC change the brightness originally?

If "Combine hardware and software dimming" and hardware DDC is  both enabled in General, then the 0-50% dimming part is software dimming - this means that the hardware part is not working. If you unclick that while you enable hardware DDC control what happens? Can you show me a screenshot of the Displays tab of MonitorControl?

You can try changing the DDC brightness using ddcctl (or m1ddc on Mac) as well or maybe try Lunar to see if it can change the displays brightness.

It is clear that the app some issues with the 27" thunderbolt display as it is quite eccentric how it works (should be controlled using Apple's protocols but also supports DDC it seems but in an unconventional way while it has no manual controls because Apple doesn't like that). Unfortunately I don't have a display like that at my disposal so I cannot experiment with it and figure out an exact procedure (or better support) as of now.

**Reply by scroggzen** [reply]
Thanks so much for the quick reply!

I was able to perform a full settings reset on Lunar - and though at first it would flash to full brightness and then revert back after about 2 seconds , I was able to make it stick by putting taking it off of manual mode and switching to sync mode.

After getting it back to full brightness, I went back to MonitorControl and as soon as I clicked the slider to move it, it immediately dimmed drastically and the same issue returned. I then closed MC and did another full reset on Lunar - flickered, and then I specifically clicked the reset brighness/contrast, and now it's back to normal.

I guess I'll just deal with the max brightness... because even though Lunar can effectively reset the settings, I can't adjust brightness.

**Reply by waydabber** [reply]
That is interesting. And you did set the DDC max value to 255 in MonitorControl beforehand and even with this at max setting the display is dimmed? I have no idea why is that. Btw can't this display be controlled natively by macOS via Thunderbolt? Or it is somehow not supported by recent macOS versions anymore?

**Reply by scroggzen** [reply]
macOS used to natively have brightness control, built-in audio, and even webcam... but they stop supporting it some time ago. Maybe High Sierra? It definitely isn't working on Mojave.

Unfortunately I had no luck with your new suggestion. I will show also include some photos of what happens with the monitor visible in the background as well.

<img width="740" alt="Screen Shot 2021-11-15 at 11 46 59 AM" src="https://user-images.githubusercontent.com/46291076/141830128-255423da-a70f-4d02-93ee-b1f5b8bf0af4.png">

You can see that I have the value set to 255 and as soon as I even click to adjust brightness, it's basically 50%:
![IMG_3086](https://user-images.githubusercontent.com/46291076/141830532-c91bf6f2-d7d6-47cc-b824-2b8eb02dce09.JPG)

And here, I open Lunar to do a brightness reset to get it back to full:
![IMG_3087](https://user-images.githubusercontent.com/46291076/141830596-151fa10a-c92f-4171-b52f-7c75a40dfcd4.JPG)

**Reply by waydabber** [reply]
Interesting. Did you disable combined dimming in General? If so, is it still changing the brightness via hardware but only up to 50%? Please disable syncing as well so that doesn't interfere in any way just to make sure (but it should  not).

Can you set the brightness manually via ddcctl https://github.com/kfix/ddcctl to 255 to get the full brightness?

(btw it is a shame that Apple stopped supporting their own display which has no actual buttons to be useful with anything other than a mac...)

**Reply by Director100DL** [reply]
I have the same problem with my iMac  and my MBP running Catalina 10.15.7. But on the MBP, Monitor Control works perfectly when hooked up to a LED Cinema Display, which is an older monitor than the Thunderbolt Display. I've set the brightness manually via ddcctl to 255 to get the full brightness.

**Reply by waydabber** [reply]
I think it would be best to simply blacklist both the Thunderbolt Display and the LED Cinema Display in the next update. Some users seem to have issues while others don't - it might be possible that there are internal firmware differences. Normally these displays were not manufactured with DDC in mind but designed to be controlled by the native Apple method via USB. They seem to be rather unreliable when controlling over DDC.

**Reply by ConlinJoe** [reply]
Tried everything and the only thing that worked for me was using ddcctl and then resetting the brightness/contrast with the -rbc flag. Tried to set the brightness to 255 and that didn't work. 

Install using Homebrew, `brew install ddcctl` and run `ddcctl -d 1 -rbc`

**Comment by dansure5** [comment]
The solution for the brightness and sound worked perfectly. I tried any values beyond 255 and in resulted in sound and brightness going down. So 255 in fact seems the perfect value.
Another issue that I found was that I can‘t get the Microphone to work with FaceTime. It works with Siri and Zoom, but not with FaceTime. Does anyone have the same issue?

**Reply by chazzzzy** [reply]
Apple did a couple system software updates since I first posted this and the ability to select the displays as the microphones disappeared whereas they show up as outputs.  

This happens for other inputs as well such as The RodeCaster Pro (in stereo, shows up in Multi Channel though).. and other inputs. RODE said it's an Apple thing.. so I would use an alternate for the time being. Unless you can set the LED DISPLAY Microphone as System input and select it that way for FaceTime. I can't remember if that works or not. Worth a try.

**Reply by dansure5** [reply]
I can choose the Display as a Input and it works with other Apps. Just FaceTime won’t work, I don’t get it. Does it work in your setup?

**Reply by chazzzzy** [reply]
It's Apple. One of the system updates recently killed various Mic inputs (including Apple's own LED display mic inputs.)

Other companies are aware of the problem and there are workarounds.. like sending the mic input into Blackhole as an aggregate input and accessing it that way.. but I think even doing that, FaceTime won't recognize it. It's really annoying, but not something anyone knows how to fix until Apple fixes it. Just use the mic input of something else for the time being until they fix it.

**Reply by dansure5** [reply]
Thank for the reply! 
I am also using the display with my iPad. There I can‘t control the volume and brightness (as expected). I hoped if I set the volume high on my Mac then it stays like that when I use the iPad, but it doesn’t seem to save the volume I set.

**Reply by dansure5** [reply]
Btw it works. With you connect it to the iPad it works with the last settings that you set on your Mac (brightness and volume level).

**Comment by rolfb** [comment]
Actually don't think you need MonitorControl for your Cinema Display at all with later versions of MacOS Monterey. Your display settings probably got bungled at some point. Try moving all files in the folder `~/Library/Preferences/ByHost` that starts with `com.apple.windowserver.displays` to a different location for backup and restart your computer. If it doesn't work, try another port on the machine. If it still doesn't work try one of the other suggestions for MonitorControl.

**Comment by scroggzen** [comment]
After dealing with the brightness issue for years (with much of that time
sadly just allowing the display to collect dust) I have found a shockingly
simple fix for my issue:

*I bought a new thunderbolt 2 cable.*

I discovered this when I decided to give it a last troubleshooting attempt,
and noticed that when I pulled up system preferences > displays, the
"brightness" slider didn't appear for the Thunderbolt Display settings. I
figured that was an odd behavior - and perhaps if that function wasn't
available in software, it might mean that the hardware simply isn't making
the connection. This led me to wonder if the degraded quality of the
thunderbolt cable would lead macOS to remove functions from system menus.
It would appear this is exactly the case.

At some point the original attached thunderbolt cable frayed just enough to
remove all *peripheral* functions of the monitor at the same time. (USB
hub, LAN, firewire, webcam, speakers) - yet still allowed the basic display
functionality (although a bit finicky). The fact that the display still
worked - albeit locked at 100% brightness - threw me off for a long time. I
figured it would either be an all-or-nothing connection, or perhaps lose
some features and not others.

Now that I am using a thunderbolt 2 cable directly from the thunderbolt
port on the display to my macbook, everything works perfectly. All
peripherals as well as software brightness controls, keyboard brightness.

The best part of this fix is that you don't have to take the monitor apart
to remove/replace the original cable. You can just leave it unplugged and
use the port. Sure, you lose 1 thunderbolt port - but that's a sacrifice
I'm happy to make!

On Fri, Sep 9, 2022 at 4:12 AM Rolf Bjaanes ***@***.***>
wrote:

> Actually don't think you need MonitorControl for your Cinema Display at
> all with later versions of MacOS Monterey. Your display settings probably
> got bungled at some point. Try moving all files in the folder
> ~/Library/Preferences/ByHost that starts with
> com.apple.windowserver.displays to a different location for backup and
> restart your computer. If it doesn't work, try another port on the machine.
> If it still doesn't work try one of the other suggestions for
> MonitorControl.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MonitorControl/MonitorControl/discussions/739#discussioncomment-3605875>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBFRBGNF3OXJDRQLXJCVZDV5L5RPANCNFSM5GWY43GA>
> .
> You are receiving this because you were mentioned.Message ID:
> <MonitorControl/MonitorControl/repo-discussions/739/comments/3605875@
> github.com>
>


-- 
Michael Scroggins
***@***.***

**Reply by dansure5** [reply]
I don’t understand the setup you used to make it work. 
Can you take a picture of it?

**Reply by scroggzen** [reply]
Pic 1: Original monitor cable obviously looking pretty rough... but it still will show a display when plugged in. (But brightness can't be adjusted, and all other functionality is gone.)
Pic 2: Showing direct thunderbolt connection to thunderbolt port on display.
Pic 3: Just showing that all of these peripherals now work. Firewire800 not pictured, but just tested before sending with an old Duet audio interface - working perfectly.

The point of this is to say - I don't think the problem I experienced (dimming to nearly unusable levels) was a direct problem with the Monitor Control software - I think that macOS removes the ability to adjust brightness values if the thunderbolt connection is degraded, even if it still has basic display functionality.

![IMG_4128](https://user-images.githubusercontent.com/46291076/190880837-6c65a859-a852-4a4d-b62b-64e85456e5bb.JPG)
![IMG_4130](https://user-images.githubusercontent.com/46291076/190880844-bbac6009-f5d7-4488-9f20-a23f447e133b.JPG)
![IMG_4129](https://user-images.githubusercontent.com/46291076/190880848-5f76b339-66de-4869-bf12-1f1aa5e75b56.JPG)

**Comment by TschonnyCache** [comment]
I was not able to talk to the display via Monitor Control nor Lunar. But I did find a solution: I used this weird little windows tool called Brightness Controler https://www.holstschumacher.dk/p/brightness-controller.html inside Windows 11 running in Parallels 18 on my MacBook Pro with M1 Pro processor.

**Comment by takeichimasato** [comment]
I am using two Cinema Display 27 monitors for Mac mini M1, one via USB-C to miniDP adaptor an the other via HDMI to DP adaptor through DP to miniDP cable. 
I cannot adjust brightness of the monitors on any tools on M1 (Venture) and resort to use a Windows PC with a DP Port. Another Windows tool called [Win 10_BrightnessSlider](https://github.com/blackholeearth/Win10_BrightnessSlider) on Windows 10 works perfect!
Unplug the monitor connector from Mac and plug it to PC and adjust brightness, and return it back to Mac.

**Comment by freshmeat777** [comment]
@TschonnyCache, @takeichimasato, hello, guys!
I have the same problem with LED Cinema Display. May be this trouble depends on miniDP adapter in Ventura?
I use this one: https://satechi.net/products/aluminum-type-c-multimedia-adapter?variant=2265695125513
But it doesn't help me with backlight. I ordered 2 another adapters and will try to take off with them. 
Which adapters do you using?

**Reply by TschonnyCache** [reply]
I‘m using the same satechi adapter. Thanks for trying out other adapters. Looking forward to your report

**Reply by masatotakeichi** [reply]
@TschonnyCache @freshmeet777 from another account here, my previous message by takeichimasato.
I am wondering too that the problem with Cinema Display is due to adaptors.
As I wrote before, I am using a USB-C to miniDP adaptor (Tunewear USB-C to miniDP TUN-OT-000045, currently unavailable)  and HDMI to DP adaptor [CableCreation HDMI to DisplayPort Adapter with USB Power](https://www.cablecreation.com/products/hdmi-to-displayport-adapter-with-usb-power) connected to Mac mini M1 (2020). Neither of them solves the problem with brightness control.
I tried out another USB-C to miniDP adaptor [MSL Force](https://www.amazon.co.jp/gp/product/B09FXLQ3CG/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1), but did not solve the problem.
I am using Mac mini with triple displays. Note that Mac mini M1 allows either of USB-C ports for display, not the both. So I use HDMI port for the second display. And for the 3rd display, another USB-A to HDMI adaptor (CableCreation) plugged in to the USB port of Cinema Display is used.

**Reply by freshmeat777** [reply]
@TschonnyCache @masatotakeichi
Colleagues, I have read on several forums, that this adapter solves problems with connecting LED Cinema Display to the last generations of Macbook Pro:
https://www.amazon.com/gp/product/B06X9J3R4V

Also, I'm ordered this one:
https://aliexpress.com/item/1005004863142912.html
which have declared compatibility with Macbook.

So, I will try this pair after it's delivering and write a report.

To this moment I have tested this adapters, but without success:
https://aliexpress.com/item/1005004342387693.html (type T4)
https://aliexpress.com/item/32967486166.html.

And, as a temporarily solution, I'm using the Brightness Controller. Thanks to @TschonnyCache! 

Cheers!

**Reply by freshmeat777** [reply]
So, as has been promised, here is the report.
I have recieved the second adapter from aliexpress. Unfortunatly it doesn't work at all.
The first adapter from amazon has not been delivered to my country due to last packet of the sanctions.
My regrets, guys, but I stay forward with windows-controlled backlight.

**Reply by masatotakeichi** [reply]
@freshmeat777 @TschonnyCache 
Thanks @freshmeat777 for your report.
A feasible solution would be to resort to Windows for the present, I think.
